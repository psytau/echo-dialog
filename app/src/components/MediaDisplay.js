const MediaDisplay = (props) => {
  const { demoMedia } = props;
  let sources = <source src={demoMedia?.url} />;
  if (demoMedia && demoMedia.url) {
    const match = demoMedia.url.match(/(.+)\.(\w+)$/);
    const url1 = match[1];
    // const ext = match[2];
    if (demoMedia.mediaType === "video") {
      sources = (
        <>
          <source src={demoMedia.url} />
          <source src={`${url1}.mp4`} />
          <source src={`${url1}.webm`} />
        </>
      );
    } else if (demoMedia.mediaType === "audio") {
      sources = (
        <>
          <source src={demoMedia.url} />
          <source src={`${url1}.mp3`} />
          <source src={`${url1}.aac`} />
          <source src={`${url1}.webm`} />
        </>
      );
    }
  }
  if (demoMedia && demoMedia.mediaType === "video" && demoMedia.url) {
    return (
      <video controls={true} className="media-demo-video">
        {sources}
      </video>
    );
  } else if (demoMedia && demoMedia.mediaType === "audio" && demoMedia.url) {
    return (
      <audio controls={true} className="media-demo-audio">
        {sources}
      </audio>
    );
  } else {
    return "";
  }
};

export default MediaDisplay;
