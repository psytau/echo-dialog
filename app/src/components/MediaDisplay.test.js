import TestRenderer from "react-test-renderer";
import MediaDisplay from "./MediaDisplay";

test("renders MediaDisplay with video", async () => {
  const demoVideoMedia = {
    mediaType: "video",
    url: "https://example.com/1234/abcd.mov",
    format: "mov",
  };
  const demoAudioMedia = {
    mediaType: "audio",
    url: "https://example.com/1234/abcd.aac",
    format: "aac",
  };
  const tree = TestRenderer.create(
    <MediaDisplay demoMedia={demoVideoMedia} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders MediaDisplay with audio", async () => {
  const demoAudioMedia = {
    mediaType: "audio",
    url: "https://example.com/1234/abcd.aac",
    format: "aac",
  };
  const tree = TestRenderer.create(
    <MediaDisplay demoMedia={demoAudioMedia} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
