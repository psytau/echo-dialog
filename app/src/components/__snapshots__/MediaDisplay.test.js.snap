// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MediaDisplay with audio 1`] = `
<audio
  className="media-demo-audio"
  controls={true}
>
  <source
    src="https://example.com/1234/abcd.aac"
  />
  <source
    src="https://example.com/1234/abcd.mp3"
  />
  <source
    src="https://example.com/1234/abcd.aac"
  />
  <source
    src="https://example.com/1234/abcd.webm"
  />
</audio>
`;

exports[`renders MediaDisplay with video 1`] = `
<video
  className="media-demo-video"
  controls={true}
>
  <source
    src="https://example.com/1234/abcd.mov"
  />
  <source
    src="https://example.com/1234/abcd.mp4"
  />
  <source
    src="https://example.com/1234/abcd.webm"
  />
</video>
`;
